certbot-apache/certbot_apache/configurator.py

```diff
@@ -972,6 +1005,10 @@ def _copy_create_ssl_vhost_skeleton(self, vhost, ssl_fp):
                 # The content does not include the closing tag, so add it
                 new_file.write("</VirtualHost>\n")
                 new_file.write("</IfModule>\n")
+            # Add new file to augeas paths if we're supposed to handle
+            # activation (it's not included as default)
+            if self.conf("handle-sites"):
+                self.parser.parse_file(ssl_fp)
```

Comment:
Before we parsed sites-available contents (added manually to augeas parse tree using value in constants.py). It's not included until the vhost is enabled by creating a symlink from sites-enabled. We need this temporarily to find the newly created vhost in the certificate configuration step.

Reply:
Calls to `parse_file` causing changes to be lost should certainly be addressed, however, if it is, perhaps we should just call this unconditionally here, especially if the logic in `make_vhost_ssl` I commented on is changed. Rather than parsing the file in two locations depending on whether or not `sites-enabled` is set, does it hurt anything to always ensure Augeas has parsed the file?